src/NUnitFramework/framework/Constraints/ComparisonAdapter.cs

```diff
@@ -167,7 +167,7 @@ public override int Compare(object expected, object actual)
                 if (!TypeHelper.TryCast(actual, out actualCast))
                     throw new ArgumentException($"Cannot compare to {actual?.ToString() ?? "null"}");
 
-                return comparison.Invoke((T)expected, (T)actual);
+                return comparison.Invoke(expectedCast, actualCast);
```

Comment:
Since we are using C# 7, can't we move the declaration of `expectedCast` and `actualCast` into the `TypeHelper.TryCast`?

Reply:
👍 I wasn't thinking, but I meant to take care of this in a future PR where I do `out var` everywhere. This commit will be overwritten by that anyway, so I'll remove it from this PR.